src/plugins/Sharesite/Freesite.java

```diff
@@ -90,6 +111,35 @@ public synchronized void setName(String name) {
 		this.name = name;
 	}
 
+	public synchronized String getPath() {
+		if (path != null) {
+			return path;
+		} else {
+			return name;
+		}
+	}
+
+	public synchronized void setPath(String path) {
+		this.path = path;
+	}
+
+	public synchronized Integer getInsertHour() {
+		if (insertHour != null) {
+			return insertHour;
+		} else {
+			return -1; // meaning unrestricted
+		}
+	}
+
+	public synchronized void setInsertHour(Integer insertHour) {
+		if (insertHour.equals(-2)) {
```

Comment:
I’m not sure what you mean here. You’re not parsing `insertHour` into anything and I have no idea whether `insertHour` being `null` is even a valid scenario here. If you want the code to treat `null` the same as `-2`, just check for both: `if ((insertHour == null) || (insertHour == -2))`.